source/Weapon.cpp

```diff
@@ -50,6 +50,8 @@ void Weapon::LoadWeapon(const DataNode &node)
 			isParallel = true;
 		else if(key == "gravitational")
 			isGravitational = true;
+		else if(key == "disregard firing hull")
```

Comment:
I feel that this shouldn't be a weapon property, but an AI personality property; Some AI ships should fire weapons that can disable or destroy themselves, while other AI ships should never fire weapons when those would destroy themselves.

Reply:
So #1428? I suppose MZ's comment is now out of date given that you can't trick a ship into using a firing hull weapon that can disable itself.

Reply:
I can see use cases for it either way. 

If we have a situationwhere such a weapon is found on a lot of ships, but only some ships are going to be sufficiently desperate to use it, then it should be a personality thing.

If it is only going to be installed on ships that are by definition desperate enough to use it, then just have the attribute on the weapon.

That being said, if the default behaviour is "do not use a weapon unless they can do so and survive," what about an attribute for "will use this weapon regardless of its effect, but only after all other secondaries are used up and/or it looks like I am about to die anyway?"

Reply:
> So #1428? I suppose MZ's comment is now out of date given that you can't trick a ship into using a firing hull weapon that can disable itself.

Yes, let's go for #1428. "being able to trick the AI" sounds to me more like a (short-term) workaround to get something done than a (long-term) desired solution.

Reply:
> That being said, if the default behaviour is "do not use a weapon unless they can do so and survive," what about an attribute for "will use this weapon regardless of its effect, but only after all other secondaries are used up and/or it looks like I am about to die anyway?"

I just posted a comment on #1428 about having more than 1 relevant personality. (#1428 seems to be the relevant ticket to discuss that).